docs/how-to-guides/develop-a-flex-flow/function-based-flow.md

```diff
@@ -0,0 +1,133 @@
+# Function based flow
+
+:::{admonition} Experimental feature
+This is an experimental feature, and may change at any time. Learn [more](../faq.md#stable-vs-experimental).
+:::
+
+User can directly use a function (see [supported types](./input-output-format.md#supported-types) for typing support) as flex flow's entry.
+
+## Authoring
+
+
+```python
+from promptflow.tracing import trace
+
+class Reply(TypedDict):
+    output: str
+
+@trace
+def my_flow(text: str) -> Reply:
+    # flow logic goes here
+    pass
+```
+
+**Note** tracing is supported for flex flow. Check [here](../tracing/index.md) for more information.
+
+## YAML support
+
+Similar as [DAG flow](../deploy-a-flow/index.md). YAML file is identifier for flex flow.
```

Comment:
There is feedback we should avoid mention flex flow , and avoid referencing flex.flow.yaml as much as possible. 

Shall we have more contents on scenario without yaml?